### ISSUE TITLE
Backend: Distinguish standalone installations from newsblur.com

### ISSUE THREAD

**Issue by da2x** [samuelclay/NewsBlur#1595]

**Comment by samuelclay** [comment]
What I'm getting from this ticket is that the publisher may want to reconfirm a subscriber? Not familiar with the need for the publisher to confirm again. Can you send me details on the spec where this is listed? I looked into some django websub apps, but nothing came up for this confirmation.

**Comment by da2x** [comment]
Newsblur sends a subscription request to a hub. This includes a secret and a callback URL. The hub then tries to confirm the subscription with a GET request to the callback url, and expects a 2xx status code. Newsblur's callback URL returns 404 instead of confirming the subscription.

It's section 5 of the webaub spec.
https://www.w3.org/TR/websub/#subscribing-and-unsubscribing

**Comment by samuelclay** [comment]
Ahh, that's right. Yes, of course this is handled, as you can see on 

https://github.com/samuelclay/NewsBlur/blob/pro/apps/push/views.py#L17-L38

But notice the url is 

https://github.com/samuelclay/NewsBlur/blob/pro/apps/push/urls.py#L5-L6

So you'd need to have `https://newsblur.com/push/<feed_id>` and that seems to be missing in your url above. Is NewsBlur sending a subscription request with a url that has no feed id?

**Comment by samuelclay** [comment]
Oh, is it possible this is why?

```         if not verify_token.startswith('subscribe'): 
             raise Http404 ```

**Comment by da2x** [comment]
So, I’ve logged 100+ failures per day since for the last 90 days. However, it seems to have stopped after 20th January. This is the last logged failure:

```
[20-Jan-2022 15:19:45 UTC] Failed to verify WebSub subscription with https://push.newsblur.com/push/14. Returned (empty) instead of the expected challenge response body. file_get_contents(https://push.newsblur.com/push/14?hub.mode=subscribe&hub.lease_seconds=1209502&hub.topic=https%3A%2F%2Ffeed.ctrl.blog%2Flatest.atom&hub.challenge=real-challange-string-here&hub.verify_token=subscribe-token-here-it-starts-with-subscribed): failed to open stream: HTTP request failed! HTTP/1.0 404 Not Found
```

The verification token starts with `subscribe`. However, the feed ID might be wrong? 14 seems low.

**Comment by samuelclay** [comment]
Ahh that 14 is the answer. That means that this is not newsblur.com but a standalone user installation. I believe I need to separate out the domains so standalone installations don't look like newsblur.com.

**Comment by samuelclay** [comment]
Fixed in 6a61620d3. Thanks for reporting this and let me know if this continues to happen.
